Core/ServeSync.Application/UseCases/EventManagement/EventCollaborationRequests/Commands/CreateEventCollaborationCommandHandler.cs

```diff
@@ -0,0 +1,71 @@
+using Microsoft.Extensions.Logging;
+using ServeSync.Application.SeedWorks.Cqrs;
+using ServeSync.Application.SeedWorks.Data;
+using ServeSync.Domain.EventManagement.EventCollaborationRequestAggregate;
+using ServeSync.Domain.EventManagement.EventCollaborationRequestAggregate.DomainServices;
+using ServeSync.Domain.EventManagement.EventOrganizationAggregate.Entities;
+using ServeSync.Domain.SeedWorks.Repositories;
+
+namespace ServeSync.Application.UseCases.EventManagement.EventCollaborationRequests.Commands;
+public class CreateEventCollaborationCommandHandler : ICommandHandler<CreateEventCollaborationCommand, Guid>
+{
+    private readonly IEventCollaborationRequestRepository _eventCollaborationRequestRepository;
+    private readonly IBasicReadOnlyRepository<EventOrganization, Guid> _eventOrganizationRepository;
+    private readonly IBasicReadOnlyRepository<EventOrganizationContact, Guid> _eventOrganizationContactRepository;
+    private readonly IEventCollaborationRequestDomainService _eventCollaborationRequestDomainService;
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly ILogger<CreateEventCollaborationCommandHandler> _logger;
+
+    public CreateEventCollaborationCommandHandler(
+        IEventCollaborationRequestRepository eventCollaborationRequestRepository,
+        IBasicReadOnlyRepository<EventOrganization, Guid> eventOrganizationRepository,
+        IBasicReadOnlyRepository<EventOrganizationContact, Guid> eventOrganizationContactRepository,
+        IEventCollaborationRequestDomainService eventCollaborationRequestDomainService,
+        IUnitOfWork unitOfWork,
+        ILogger<CreateEventCollaborationCommandHandler> logger)
+    {
+        _eventCollaborationRequestRepository = eventCollaborationRequestRepository;;
+        _eventOrganizationRepository = eventOrganizationRepository;
+        _eventOrganizationContactRepository = eventOrganizationContactRepository;
+        _eventCollaborationRequestDomainService = eventCollaborationRequestDomainService;
+        _unitOfWork = unitOfWork;
+        _logger = logger;
+    }
+    public async Task<Guid> Handle(CreateEventCollaborationCommand request, CancellationToken cancellationToken)
+    {
+        await _unitOfWork.BeginTransactionAsync();
```

Comment:
This is UnitOfWork implemented by database, when use this function you should end up with CommitTransactionAsync or that transaction will never be closed and your changes nerver be updated
Note: Use this when EfCore SaveChanges needed tasks that commit to db but wrapped by db transaction